README.md

```diff
@@ -40,7 +40,83 @@ npm install and-cli -g
+├── utilities/                              # Utility functions that aren't really categorized as a standard 'module'
+|   ...
+|   ├── option-string-factory.js            # Factory for building out option strings to be passed to `program.option`, ie `-i, --info`
+|   └── option-string-factory.test.js       # Unit tests file for the option-string-factory utility
+├── and-cli.js                              # Main entrypoint/parent command that registers subcommands
+├── and-cli-copy.js                         # Implementation of the 'copy' command
+├── and-cli-copy.test.js                    # Integration test file for the 'copy' command
+| ...
+├── and-cli.test.js                         # Integration test file for the main entrypoint/parent command
+├── COMMANDS.md                             # Markdown file containing extra documentation for each command
+├── command-runner.js                       # Utility module for wrapping command body functions in to mimic top-level async
+├── package.json
+└── package-lock.json
+
+```
+
+## Testing strategy
+
+For testing, we use [`Jest`](https://github.com/facebook/jest). We integration test top-level commands (such as `and-cli-dotnet`) and unit test shared modules/utilities (such as `modules/dotnet-build` or `utilities/option-string-factory`).
+
+While we do not have any coverage thresholds currently configured, we ask that new modules/utilities introduced to the codebase are unit tested for high-value paths. Integration tests should be considered, but written more sparingly due to the overhead of run-time.
+
+## Extending functionality
+
+The functionality of this CLI can be extended by adding it as a dependency in your node project and requiring the main module, ie `require("and-cli")`. All commands should be registered through the `command-registry` module, which provides multiple functions for adding, overriding, and removing commands.
+
+A small example of a project that imports this package and pulls in all of the base commands:
+
+```JS
+#!/usr/bin/env node
+
+// -----------------------------------------------------------------------------------------
+// #region Imports
+// -----------------------------------------------------------------------------------------
+
+const commandRegistry = require("and-cli/modules/command-registry");
+const program = require("and-cli");
+
+// #endregion Imports
+
+// -----------------------------------------------------------------------------------------
+// #region Entrypoint
+// -----------------------------------------------------------------------------------------
+
+// Register all of the base commands from the and-cli with this application
+commandRegistry.registerBaseCommands();
+
+program.parse(process.argv);
+
+// #endregion Entrypoint
+```
+
+For more examples, or to see what the full project structure might look like, see this example repository:
+
+[`and-cli-plugin-example`](https://github.com/brandongregoryscott/and-cli-plugin-example)
```

Comment:
Should this repo be brought into AndcultureCode?

Reply:
It definitely could be, just needed something to quick iterate on while developing 👍 Thoughts on naming? Maybe `AndcultureCode.Cli.Example` or `AndcultureCode.Cli.Boilerplate`?

Reply:
How about `AndcultureCode.Cli.PluginExample` just so it's specific about being related to the plugin portion of the CLI?

Reply:
Imported the example repo here: https://github.com/AndcultureCode/AndcultureCode.Cli.PluginExample I'll update the link here, and add a PR there for some minor cleanup (old repo/branch references, license, etc.)